gravatar/src/main/java/com/gravatar/services/ProfileService.kt

```diff
@@ -29,23 +30,29 @@ public class ProfileService(private val okHttpClient: OkHttpClient? = null) {
      * @param hashOrUsername The hash or username to fetch the profile for
      * @return The fetched profile or null if profile not found
      */
-    public suspend fun retrieve(hashOrUsername: String): Profile? = withContext(GravatarSdkDI.dispatcherIO) {
-        val response = service.getProfileById(hashOrUsername)
-        if (response.isSuccessful) {
-            response.body() ?: error("Response body is null")
-        } else {
-            // Log the response body for debugging purposes if the response is not successful
-            Logger.w(
-                LOG_TAG,
-                "Network call unsuccessful trying to get Gravatar profile: ${response.code()}",
-            )
-            if (response.code() == HttpResponseCode.HTTP_NOT_FOUND) {
-                return@withContext null
-            } else {
-                throw HttpException(response)
+    public suspend fun retrieve(hashOrUsername: String): Profile? =
+        @Suppress("TooGenericExceptionCaught")
+        try {
+            withContext(GravatarSdkDI.dispatcherIO) {
+                val response = service.getProfileById(hashOrUsername)
+                if (response.isSuccessful) {
+                    response.body() ?: error("Response body is null")
+                } else {
+                    // Log the response body for debugging purposes if the response is not successful
+                    Logger.w(
+                        LOG_TAG,
+                        "Network call unsuccessful trying to get Gravatar profile: ${response.code()}",
+                    )
+                    if (response.code() == HttpResponseCode.HTTP_NOT_FOUND) {
+                        return@withContext null
+                    } else {
+                        throw HttpException(response)
+                    }
+                }
             }
+        } catch (ex: Exception) {
+            throw GravatarException(ex.errorType(GravatarSdkContainer.instance.moshi), ex)
         }
-    }
```

Comment:
Is there a reason why we can't use `runThrowingExceptionRequest` here?

Reply:
In this case, the returned value when 404 is null instead of the exception, so we can't use `runThrowingExceptionRequest`, which requires not null.